snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -470,9 +470,36 @@ def prime(self, force=False) -> None:
                 _migrate_files(system, system_dependency_paths, '/',
                                self.primedir, follow_symlinks=True)
 
+        # TODO: base snap support
+        core_path = common.get_core_path()
+
+        def library_pather(library_path: str, elf_file_path: str) -> str:
+            # If the path is is in the core snap, use the absolute path,
+            # if the path is primed, use $ORIGIN, and last if the dependency
+            # is not anywhere return an empty string.
+            #
+            # Once we move away from the system library grabbing logic
+            # we can move to a smarter library capturing mechanism.
+            library_path = library_path.replace(self.installdir, self.primedir)
+            if library_path.startswith(core_path):
+                return library_path
+            elif (library_path.startswith(self.primedir) and
+                  os.path.exists(self.primedir)):
+                rel_library_path = os.path.relpath(library_path, elf_file_path)
+                rel_library_path_dir = os.path.dirname(rel_library_path)
+                # return the dirname, with the first .. replace with $ORIGIN
+                return rel_library_path_dir.replace('..', '$ORIGIN', 1)
```

Comment:
This doesn't sound quite right. Doesn't `$ORIGIN` mean "this file"? It sounds like you want this path prepended with it, not using it to replace `..`.

Reply:
I thought so to, originally, but no, after testing, this is what I want; if you play with relpath you will see what I mean.

Reply:
Here's an example of the current behaviour:
```
(snapcraft) ubuntu@snapcraft-dev:~/git$ ldd /home/ubuntu/git/prime/libexec/git-core/git-imap-send|grep curl
	libcurl.so.4 => /home/ubuntu/git/prime/libexec/git-core/../../usr/lib/x86_64-linux-gnu/libcurl.so.4 (0x00007f4dab7bb000)
(snapcraft) ubuntu@snapcraft-dev:~/git$ ls /home/ubuntu/git/prime/libexec/git-core/../../usr/lib/x86_64-linux-gnu/libcurl.so.4
/home/ubuntu/git/prime/libexec/git-core/../../usr/lib/x86_64-linux-gnu/libcurl.so.4
```